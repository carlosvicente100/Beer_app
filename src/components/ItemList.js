import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
//import dummyList from "../fixtures/beerList";

import Card from "./Card";

const ItemListContainer = styled.div`
  // background-color: #9b59b6;
  display: grid;
  grid-template-columns: auto auto;
  margin: auto;
  column-gap: 20px;
  row-gap: 20px;
  margin-inline-start: 20px;
  margin-inline-end: 20px;
`;

const ItemList = ({ items }) => {
  return (
    <ItemListContainer className="ItemList">
      {items.length > 0 ? (
        items.map(item => <Card key={item.id} {...item} />)
      ) : (
        <div>no beers for you :(</div>
      )}
    </ItemListContainer>
  );
};

ItemList.propTypes = {
  items: PropTypes.array.isRequired
};
ItemList.defaultProps = {};

export default ItemList;
