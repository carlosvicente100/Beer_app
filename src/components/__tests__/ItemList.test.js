import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import ItemList from '../ItemList'
import fixtureItemList from '../../fixtures/itemList'

describe('ItemList Component Suite', () => {
  test('Snapshot check', () => {
    const component = renderer.create(<ItemList items={fixtureItemList} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should show each item correcty - array with 3 values', () => {
    const { getByText } = render(<ItemList items={fixtureItemList} />)
    expect(getByText(fixtureItemList[0].name)).toBeInTheDocument()
    expect(getByText(fixtureItemList[1].name)).toBeInTheDocument()
    expect(getByText(fixtureItemList[2].name)).toBeInTheDocument()
  })

  test('Should use props correcty - empty array', () => {
    const { getByText } = render(<ItemList items={[]} />)
    expect(getByText('no beers for you :(')).toBeInTheDocument()
  })
})
