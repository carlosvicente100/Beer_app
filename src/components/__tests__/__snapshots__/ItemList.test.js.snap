// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component Suite Snapshot check 1`] = `
.c1 {
  background-color: #706fd3;
  color: black;
  border-radius: 25px 0 25px 0;
  overflow: hidden;
  height: 200px;
  display: grid;
  grid-template-rows: 50px 20px auto;
  grid-template-columns: 125px auto auto;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  grid-template-areas: 'image title title' 'image . .' 'image description description';
}

.c1 img {
  grid-area: image;
}

.c1 h1 {
  height: auto;
  grid-area: title;
  font-size: 16px;
  text-transform: uppercase;
}

.c1 p {
  grid-area: description;
  font-size: 14px;
}

.c2 {
  margin-inline-start: 20px;
  height: 200px;
  width: auto;
}

.c0 {
  display: grid;
  grid-template-columns: auto auto;
  margin: auto;
  -webkit-column-gap: 20px;
  column-gap: 20px;
  row-gap: 20px;
  margin-inline-start: 20px;
  margin-inline-end: 20px;
}

<div
  className="c0 ItemList"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://images.punkapi.com/v2/keg.png"
    />
    <h1>
      Buzz
    </h1>
    <span>
      A Real Bitter Experience.  (09/2007)
    </span>
    <p>
      A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
    </p>
  </div>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://images.punkapi.com/v2/2.png"
    />
    <h1>
      Trashy Blonde
    </h1>
    <span>
      You Know You Shouldn't  (04/2008)
    </span>
    <p>
      A titillating, neurotic, peroxide punk of a Pale Ale. Combining attitude, style, substance, and a little bit of low self...
    </p>
  </div>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://images.punkapi.com/v2/keg.png"
    />
    <h1>
      Berliner Weisse With Yuzu - B-Sides
    </h1>
    <span>
      Japanese Citrus Berliner Weisse.  (11/2015)
    </span>
    <p>
      Japanese citrus fruit intensifies the sour nature of this German classic.
    </p>
  </div>
</div>
`;
